Modules/@babylonjs/react-native/android/src/main/java/com/babylonreactnative/EngineView.java

```diff
@@ -114,13 +168,20 @@ public void takeSnapshot() {
         helperThread.start();
         final Handler helperThreadHandler = new Handler(helperThread.getLooper());
 
-        SurfaceView surfaceView = this.primarySurfaceView;
+        Surface sourceSurface = this.transparentSurface;
         if (BabylonNativeInterop.isXRActive()) {
-            surfaceView = this.xrSurfaceView;
+            sourceSurface = this.xrSurfaceView.getHolder().getSurface();
+        } else if (this.opaqueSurfaceView != null) {
+            sourceSurface = this.opaqueSurfaceView.getHolder().getSurface();
         }
+        PixelCopy.request(sourceSurface, bitmap, getOnPixelCopyFinishedListener(bitmap, helperThread), helperThreadHandler);
+    }
 
-        // Request the pixel copy.
-        PixelCopy.request(surfaceView, bitmap, (copyResult) ->  {
+    // ---------------------------------------------------------------------------------------------
+    // Returns the listener for the PixelCopy.request function call
+    @NonNull
+    private PixelCopy.OnPixelCopyFinishedListener getOnPixelCopyFinishedListener(Bitmap bitmap, HandlerThread helperThread) {
```

Comment:
Do we really need a function that returns a function for this? Why not just leave it inline as it originally was?

Reply:
I find it more clear that way.